src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -301,68 +317,61 @@ private static string DecodePropertyName(string name, string language)
             return name;
         }
 
-        private class DeclarationGenerator : SymbolVisitor
+        /// <summary>
+        /// The <see langword="bool"/> type parameter indicates if we succeeded at writing out the declaration ID or not. 
+        /// </summary>
+        private sealed class PrefixAndDeclarationGenerator(StringBuilder builder) : SymbolVisitor<bool>
         {
-            private readonly StringBuilder _builder;
-            private readonly Generator _generator;
-
-            public DeclarationGenerator(StringBuilder builder)
-            {
-                _builder = builder;
-                _generator = new Generator(builder);
-            }
+            private readonly StringBuilder _builder = builder;
+            private readonly DeclarationGenerator _generator = new DeclarationGenerator(builder);
 
-            public override void DefaultVisit(ISymbol symbol)
-            {
-                throw new InvalidOperationException("Cannot generated a documentation comment id for symbol.");
-            }
+            /// <summary>
+            /// If we hit anything we don't know about, indicate failure.
+            /// </summary>
+            public override bool DefaultVisit(ISymbol symbol)
+                => false;
```

Comment:
>=> false; [](http://example.com/codeflow?start=16&length=9)

This change feels fragile, there is no guarantee that every caller is going to check the return value and propagate `false`.

Reply:
there is only two external caller.  One of which is the mainline call (And checks) and the other is a caller within referencegenerator which checks as well.  It doesn't feel particularly fragile to me.  No more so than any case where a visitor (or any api) is used and has a return value.  I can doc the type even more heavily to indicate that it might fail if that helps.

Reply:
Moved to a non-flow based approach to this determination.